lib/vagrant_cloud/cli.rb

```diff
@@ -0,0 +1,71 @@
+require 'thor'
+
+module VagrantCloud
+  class Cli < Thor
+    package_name 'VagrantCloud'
+
+    class_option :username, alias: '-u', required: true, desc: 'Vagrant Cloud username'
+    class_option :token, alias: '-t', required: true, desc: 'Vagrant Cloud access token'
+    class_option :box, alias: '-b', required: true, desc: 'name of the box'
+    class_option :verbose, type: :boolean
+
+    desc 'create_version', 'creates a new version within a given box'
+    method_option :version, alias: '-v', required: true, desc: 'version within the box'
+    def create_version
+      version = current_box.create_version(options[:version])
+      puts "created #{version.version} of box #{options[:box]}; current status is #{version.status}" if options[:verbose]
+      version
+    end
+
+    desc 'release_version', 'release the specified version within a given box'
+    method_option :version, alias: '-v', required: true, desc: 'version within the box'
+    def release_version
+      version = current_version
+      version.release
+      puts "Box #{options[:box]} / version: #{version.version}; current status is #{version.status}" if options[:verbose]
+      true
+    end
+
+    desc 'create_provider', 'creates a provider within a given box and version'
+    method_option :version, alias: '-v', required: true, desc: 'version within the box'
+    method_option :provider, alias: '-p', default: 'virtualbox', desc: 'the provider for the box; default: virtualbox'
+    method_option :provider_url, alias: '-pu', desc: 'URL to the file for remote hosting; either a _url or _file_path can be provided, not both'
```

Comment:
The reference to th "_file_path" is probably not needed here, because it's only used in the *other* function below (`upload_file`)?